src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -0,0 +1,201 @@
+      sourceClientFactory = getSourceClientFactory(fileLoadModel.getSourcePath());
+    } else {
+      // Use application level authentication
+      sourceClientFactory =
+          getSourceClientFactory(
+              blobUrl.getAccountName(),
+              resourceConfiguration.getAppToken(UUID.fromString(profileModel.getTenantId())),
+              blobUrl.getBlobContainerName());
+    }
+
+    BlobCrl blobCrl = getBlobCrl(destinationClientFactory);
+
+    // Read the leaf node of the source file to use as a way to name the file we store
+    String fileName = getLastNameFromPath(blobUrl.getBlobName());
+
+    String blobName = getBlobName(fileId, fileName);
+    blobCrl
+        .createBlobContainerCopier(sourceClientFactory, blobUrl.getBlobName(), blobName)
+        .beginCopyOperation()
+        .waitForCompletion();
+
+    BlobProperties blobProperties = blobCrl.getBlobProperties(blobName);
+    Instant createTime = blobProperties.getCreationTime().toInstant();
+    return new FSFileInfo()
+        .fileId(fileId)
+        .createdDate(createTime.toString())
+        .gspath(
+            String.format(
+                "%s/%s",
+                destinationClientFactory.getBlobContainerClient().getBlobContainerUrl(), blobName))
+        .checksumMd5(Base64.getEncoder().encodeToString((blobProperties.getContentMd5())))
+        .size(blobProperties.getBlobSize())
+        .bucketResourceId(storageAccountResource.getResourceId().toString());
+  }
+
+  public boolean deleteDataFileById(
+      String fileId, String fileName, AzureStorageAccountResource storageAccountResource) {
+    BillingProfileModel profileModel =
+        profileDao.getBillingProfileById(storageAccountResource.getProfileId());
+
+    BlobContainerClientFactory destinationClientFactory =
+        getTargetDataClientFactory(profileModel, storageAccountResource, true);
+    String blobName = getBlobName(fileId, fileName);
+    BlobCrl blobCrl = getBlobCrl(destinationClientFactory);
+    try {
+      blobCrl.deleteBlob(blobName);
+      return true;
+    } catch (BlobStorageException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+        return false;
+      } else {
+        throw new PdaoException("Error deleting file", e);
+      }
+    }
+  }
+
+  public boolean deleteFile(FireStoreFile fireStoreFile) {
+    AzureStorageAccountResource storageAccountResource =
+        azureResourceDao.retrieveStorageAccountById(
+            UUID.fromString(fireStoreFile.getBucketResourceId()));
+
+    BillingProfileModel profileModel =
+        profileDao.getBillingProfileById(storageAccountResource.getProfileId());
+    BlobContainerClientFactory destinationClientFactory =
+        getTargetDataClientFactory(profileModel, storageAccountResource, true);
+
+    BlobUrlParts blobParts = BlobUrlParts.parse(fireStoreFile.getGspath());
+    if (!StringUtils.equals(blobParts.getAccountName(), storageAccountResource.getName())) {
```

Comment:
Is it possible for `accountName` to be `null`? If not, you can use `String.equals()` here.

Reply:
Gah, good point! It is not